### DISCUSSION TITLE
After upgrading Python, is 'sudo apt autoremove' off-limits?

### DISCUSSION THREAD

**Discussion by dave-culp** [CloudBytes-Academy/CloudBytes.dev#114] [Q&A]
In this thread (https://cloudbytes.dev/snippets/upgrade-python-to-latest-version-on-ubuntu-linux), it is stated:

"DO NOT RUN sudo apt autoremove as it will remove several packages that are required. This may break your system if you're using GUI, if you're on WSL2 you can proceed."

Does this mean that once we complete this upgrade to the latest Python version, the "sudo apt autoremove" command should never be used again?

Thanks,
Dave

![Screenshot from 2022-05-29 23-44-05](https://user-images.githubusercontent.com/73796625/170925287-cafdd473-8cae-467c-ae49-bffabb94316a.png)

**Comment by rehanhaider** [comment]
The answer is a bit tricky, but let me try
1. It is not off-limits forever, but
2. The impact is a bit random as the issue is not affecting all users
3. Usually after installing a few other packages & running updates it will hold

Keep in mind there is no need for you to upgrade Python and this is just for lazy devs like me who can recreate their dev environment at will and don't like using multiple versions of Python.
